commands/scp.go

```diff
@@ -148,6 +150,11 @@ func getInfoForScpArg(hostAndPath string, hostInfoLoader HostInfoLoader) (HostIn
 	}
 
 	args := []string{}
+	port, err := hostInfo.GetSSHPort()
+	if err == nil && port > 0 {
+		args = append(args, "-P", fmt.Sprintf("%v", port))
+	}
```

Comment:
I think it should be done in `generateLocationArg()` instead of `getInfoForScpArg()`. Your unit test below is generating the following : 
```
	expectedArgs := append(
		baseSSHArgs,
		"-3",
		"-r",
		"-o",
		"IdentitiesOnly=yes",
		"-P",
		"234",
		"-i",
		"/fake/keypath/id_rsa",
		"/tmp/foo",
		"root@12.34.56.78:/home/docker/foo",
	)
```

While it seems like it should be : 
```
	expectedArgs := append(
		baseSSHArgs,
		"-3",
		"-r",
		"-o",
		"IdentitiesOnly=yes",
		"-i",
		"/fake/keypath/id_rsa",
		"/tmp/foo",
		"-P",
		"234",
		"root@12.34.56.78:/home/docker/foo",
	)
```

So that in the case of a transfer with 2 remote machines using non standard ports, each -P flag is put in front of the appropriate location.